lib/chef/resource/aws_cloudsearch_domain.rb

```diff
@@ -0,0 +1,46 @@
+require 'chef/provisioning/aws_driver/aws_resource'
+
+module AWS
+  class CloudSearch
+    class Domain
+      # The version of the AWS sdk we are using doesn't have a model
```

Comment:
I think it is better to keep this for now.  My problem with using an existing class is that future code might make assumptions about this being an actual AWS model object, and if it does I want it to blow up loudly.  If we use another object in the system we don't control, it might silently "work" for some time and produce hard-to-debug bugs.